Sources/NIOConcurrencyHelpers/lock.swift

```diff
@@ -160,6 +204,28 @@ public final class ConditionLock<T: Equatable> {
     public func lock(whenValue wantedValue: T, timeoutSeconds: Double) -> Bool {
         precondition(timeoutSeconds >= 0)
 
+#if os(Windows)
+        var dwMilliseconds: DWORD = DWORD(timeoutSeconds * 1000)
+        while true {
+          if self._value == wantedValue {
```

Comment:
I think this is reading `_value` without a lock which is undefined behaviour

Reply:
Well, thats not the only issue - the mutex was never locked ... this is missing a `self.lock`.

Reply:
yep, that's what I mean :)

Reply:
Nice catch!  I may have to ask you to do this once more ... the MSVCRT port for System.swift is ... challenging.